ui/src/hooks/helper/index.ts

```diff
@@ -124,19 +125,31 @@ export const handleCustomKey = (
   lunaId: string,
   origin: string
 ): boolean => {
+
+  const debouncedSwitchTab = useDebounceFn((lunaId: string, origin: string, key: string) => {
+    switch (key) {
+      case 'ArrowRight':
+        sendEventToLuna('KEYEVENT', 'alt+shift+right', lunaId, origin);
+        break;
+      case 'ArrowLeft':
+        sendEventToLuna('KEYEVENT', 'alt+shift+left', lunaId, origin);
+        break;
+    }
+  }, 500);
+
   if (e.altKey && e.shiftKey && (e.key === 'ArrowRight' || e.key === 'ArrowLeft')) {
     switch (e.key) {
       case 'ArrowRight':
         if (lunaId && origin) {
-          sendEventToLuna('KEYEVENT', 'alt+shift+right', lunaId, origin);
+          debouncedSwitchTab(lunaId, origin, 'ArrowRight');
         } else {
           mittBus.emit('alt-shift-right');
         }
 
         break;
       case 'ArrowLeft':
         if (lunaId && origin) {
-          sendEventToLuna('KEYEVENT', 'alt+shift+left', lunaId, origin);
+          debouncedSwitchTab(lunaId, origin, 'ArrowLeft');
         } else {
           mittBus.emit('alt-shift-left');
         }
```

Comment:
The code has no issues but there is some inconsistency with variable names:

1. `lunaId: string` should ideally be `lunaConfig: ILunaConfig`
2. In line 38, "handleCustomKey" could be renamed to something easier like "switchHandler". 

Additionally, since you're currently not handling errors on the console, I might suggest adding them for readability.
Please note that this suggestion pertains only to naming changes and reformatting of your code for simplicity and clarity. The functionality remains unchanged.

In terms of optimization, it may help to reduce unnecessary variables (e.g., remove unused functions). This will depend on specific performance metrics and debugging results.